src/dbnode/persist/fs/msgpack/encoder.go

```diff
@@ -53,21 +53,32 @@ type Encoder struct {
 	legacy legacyEncodingOptions
 }
 
+type legacyEncodingIndexInfoVersion int
+
+const (
+	// List in reverse order to ensure default value is current version.
+	legacyEncodingIndexVersionCurrent legacyEncodingIndexInfoVersion = iota
+	legacyEncodingIndexVersionV2
+	legacyEncodingIndexVersionV1
```

Comment:
To avoid Current being the default value, just create a const that is the Default that points to the current?

The opposite direction is just so much more natural and clean (avoiding capturing things like the default value being the zero value, which seems brittle). 

```go
const (
  legacyEncodingIndexInfoVersionCurrent = legacyEncodingIndexVersionCurrent

  legacyEncodingIndexInfoVersionV1 legacyEncodingIndexInfoVersion = iota
  legacyEncodingIndexInfoVersionV2
  legacyEncodingIndexInfoVersionV3 
)
```